modules/tensor_mechanics/src/transfers/VectorPostprocessorTransfer.C

```diff
@@ -0,0 +1,299 @@
+//* This file is part of the MOOSE framework
```

Comment:
A few thoughts on this transfer:

1. This seems like something that may have general utility in MOOSE beyond tensor_mechanics appliations, and should probably be put in the framework instead. It looks like there already is a MultiAppVectorPostprocessorTransfer in MOOSE that transfers from a VPP in the subapp to a VPP in the master app.
2) The special feature of this transfer is that this transfers to an AuxVariable rather than a VPP. Do you think there are applications of this to problems that might not just have line elements like our problem of interest?
3) I think this doesn't actually work with any VPP in general -- it has to work with VPPs that inherit from SamplerBase. You should probably check for that. Also, those samplers all output a (poorly named) vector called 'id' which gives the position along the line, so if you used that for your coordinate, you would not need `subapp_component`, and that component would not be restricted to the Cartesian axes.
4) There's also an argument for not having the `master_component` restricted to the Cartesian axes. You could have the user supply a vector the gives the direction in the master app for the VPP quantities to be interpolated along.